src/expressions.rs

```diff
@@ -36,7 +33,12 @@ fn find_closest_state(inputs: &[Series]) -> PolarsResult<Series> {
             for (lhs_opt_val, rhs_opt_val) in lhs_arr.iter().zip(rhs_arr.iter()) {
                 match (lhs_opt_val, rhs_opt_val) {
                     (Some(lhs_val), Some(rhs_val)) => {
-                        let res = &geocoder.search((*lhs_val, *rhs_val)).record.admin1;
+                        let record = &geocoder.search((*lhs_val, *rhs_val)).record;
+                        let res: &String = match location_level {
+                            LocationLevel::City => &record.name,
```

Comment:
does it not work to use `binary_elementwise_into_string_amortized` in this function, like we (used to) do in the `reverse_geocode` function? then it should be a lot simpler

Reply:
it should be the same, IIRC `binary_elementwise_into_string_amortized` is a convenience function for the longer version